match_realtime.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python
+
+'''
+Copyright (c) 2021 Stephen Hausler, Sourav Garg, Ming Xu, Michael Milford and Tobias Fischer
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+
+Extracts Patch-NetVLAD local and NetVLAD global features from a given directory of images.
+
+Configuration settings are stored in configs folder, with compute heavy performance or light-weight alternatives
+available.
+
+Features are saved into a nominated output directory, with one file per image per patch size.
+
+Code is dynamic and can be configured with essentially *any* number of patch sizes, by editing the config files.
+'''
+
+
+import argparse
+import configparser
+import os
+from os.path import join, exists, isfile
+
+from tqdm.auto import tqdm
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+import numpy as np
+import cv2
+from PIL import Image
+from matplotlib import pyplot as plt
+
+
+from patchnetvlad.models.models_generic import get_backend, get_model, get_pca_encoding
+from patchnetvlad.tools.patch_matcher import PatchMatcher
+from patchnetvlad.models.local_matcher import calc_keypoint_centers_from_patches as calc_keypoint_centers_from_patches
+from patchnetvlad.tools import PATCHNETVLAD_ROOT_DIR
+
+
+def input_transform(resize=(480, 640)):
```

Comment:
Why not `from patchnetvlad.tools.datasets import input_transform`?

Reply:
Sounds good, ill do that